### ISSUE TITLE
[core][observability] Ray Dashboard per process GRAM usage measures utilization and not allocated memory

### ISSUE THREAD

**Issue by nikitavemuri** [ray-project/ray#61456]
### What happened + What you expected to happen

- Ray dashboard per process GRAM is computed from `pynvml.nvmlDeviceGetProcessesUtilizationInfo().memUtil`. 
- This represents the memory bus utilization (`Percent of time over the past sample period during which global (device) memory was being read or written.` from https://docs.nvidia.com/deploy/nvml-api/structnvmlUtilization__t.html#structnvmlUtilization__t%5B/url%5D), and not how much memory was allocated. 
- As a result, there's a discrepancy between the GRAM usage reported on Ray Dashboard and the `ray_node_gram_used` metric emitted for Prometheus scraping.
https://github.com/ray-project/ray/blob/edefe438250147e6a4012645970a73675614281a/python/ray/dashboard/modules/reporter/gpu_providers.py#L329-L343

### Versions / Dependencies

Ray 2.54 (or latest)

### Reproduction script

```
import os
import time
import ray
import torch

ray.init(address="auto")

@ray.remote(num_gpus=1)
class GpuMemActor:
    def __init__(self, gib=8):
        self.gib = gib
        self.physical_gpu_index = ray.get_runtime_context().get_accelerator_ids()["GPU"][0]
        n = int(gib * (1024**3) / 4)
        self.buf = torch.empty(n, dtype=torch.float32, device="cuda")
        self.buf.fill_(1.0)

    def info(self):
        return {
            "pid": os.getpid(),
            "host": os.uname().nodename,
            "gpu_index": self.physical_gpu_index,
            "allocated_mib": torch.cuda.memory_allocated() / (1024**2),
            "reserved_mib": torch.cuda.memory_reserved() / (1024**2),
        }

    def burn(self, seconds=1200):
        end = time.time() + seconds
        while time.time() < end:
            self.buf.mul_(1.000001)
            torch.cuda.synchronize()
        return True

a = GpuMemActor.options(name="gpu-mem-repro").remote(gib=8) 
print("Actor info:", ray.get(a.info.remote()))
a.burn.remote(1200)

while True:
    print(ray.get(a.info.remote()))
    time.sleep(30)
```

<img width="1000" height="834" alt="Image" src="https://github.com/user-attachments/assets/88bba72d-cfda-4888-8152-5bfe8ac09b4d" />

<img width="1000" height="1480" alt="Image" src="https://github.com/user-attachments/assets/9bcaab56-15be-424d-a00b-c25f5dd58587" />

### Issue Severity

P1
